src/main/java/com/greenfox/tribes/services/EquipmentService.java

```diff
@@ -20,6 +23,11 @@ public Equipment randomEquipment() {
     return equipmentList.get(index);
   }
 
+  public CharacterEquipment findCharacterEquipment(Long id) {
+
+    return characterEquipmentRepository.findById(id).orElse(null);
```

Comment:
Return optional here, this puts you in danger of forgetting about the null and then getting null pointer exceptions